Nodejs/Product/Nodejs/TestFrameworks/mocha/mocha.js

```diff
@@ -53,33 +57,68 @@ var run_tests = function (testName, testFile, workingFolder, projectFolder) {
         return;
     }
 
-    var mocha = new Mocha();
-    mocha.ui('tdd');
-    //set timeout to 10 minutes, because the default of 2 sec might be too short (TODO: make it configurable)
-    mocha.suite.timeout(600000);
+    var mocha = initializeMocha(Mocha, projectFolder);
+
     if (testName) {
         mocha.grep(testName);
     }
     mocha.addFile(testFile);
 
-    // Choose 'tap' rather than 'min' or 'xunit'. The reason is that
-    // 'min' produces undisplayable text to stdout and stderr under piped/redirect, 
-    // and 'xunit' does not print the stack trace from the test.
-    mocha.reporter('tap');
-
     mocha.run(function (code) {
         process.exit(code);
     });
 };
 
+function logError(errorMessage) {
+    console.log("NTVS_ERROR: " + errorMessage);
+}
+
 function detectMocha(projectFolder) {
     try {
         var Mocha = new require(projectFolder + '\\node_modules\\mocha');
         return Mocha;
     } catch (ex) {
-        console.log("NTVS_ERROR:Failed to find Mocha package.  Mocha must be installed in the project locally.  Mocha can be installed locally with the npm manager via solution explorer or with \".npm install mocha\" via the Node.js interactive window.");
+        logError("Failed to find Mocha package.  Mocha must be installed in the project locally.  Mocha can be installed locally with the npm manager via solution explorer or with \".npm install mocha\" via the Node.js interactive window.");
         return null;
     }
 }
 
+function initializeMocha(Mocha, projectFolder) {
+    var mocha = new Mocha();
+    applyMochaOptions(mocha, getMochaOptions(projectFolder));
+    return mocha;
+}
+
+function applyMochaOptions(mocha, options) {
+    if (options) {
+        for (var opt in options) {
+            var mochaOpt = mocha[opt];
+            var optValue = options[opt];
+
+            if (typeof mochaOpt == 'function') {
+                try {
+                    mochaOpt.call(mocha, optValue);
+                } catch (e) {
+                    logError("Could not set mocha option '" + opt + "' with value '" + optValue + "' due to error: " + e);
+                }
+            }
+        }
+    }
+}
+
+function getMochaOptions(projectFolder) {
+    var mochaOptions = defaultMochaOptions;
+    try {
+        var options = require(projectFolder + '\\test\\mocha.json');
+        options = options || {};
+        for (var opt in options) {
+            mochaOptions[opt] = options[opt];
+        }
+    } catch (ex) {
+        console.log("NTVS: mocha.json options file not found. Using default values.");
```

Comment:
I saw your comment fly by in email, but I can't see it on github for some reason. Anyways to elaborate on this... as the code is written now, the user will not see it. "NTVS_ERROR" is a string that we search for and strip out of the final output when we display it in the output window. 
``` csharp
IEnumerable<String> stdoutLines = stdout.Split(new string[] {Environment.NewLine},
    StringSplitOptions.RemoveEmptyEntries).Where(s => s.StartsWith("NTVS_ERROR:")).Select(s => s.Trim().Remove(0,11));
```
https://github.com/Microsoft/nodejstools/blob/35817ffdaa1797f95628a22f588fc3ed885f884b/Nodejs/Product/TestAdapter/TestFrameworks/TestFramework.cs#L53-L54

You'll only see the full output printed out when the product is built with the "Debug" configuration, but not the "Release" config. So using ```logError``` is the way to go here.